File: src/leetcode/number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.js
/**
 * Leetcode #1689
 * Difficulty: Easy
 * https://leetcode.com/problems/partitioning-into-minimum-number-of-deci-binary-numbers/
 */
/**
 * @param {number[][]} rectangles
 * @return {number}
 */
export const countGoodRectangles = rectangles => {
  let max = 0;
  let count = 0;

  for (const [l, w] of rectangles) {
    const size = Math.min(l, w);

    if (size > max) {
      max = size;
      count = 1;
    } else if (size === max) {
      count += 1;
    }
  }

  return count;
};

File: src/leetcode/number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.rs
impl Solution {
    pub fn count_good_rectangles(rectangles: Vec<Vec<i32>>) -> i32 {
        let mut count = 0;
        let mut max = 0;

        for rectangle in rectangles {
            let size = i32::min(rectangle[0], rectangle[1]);

            if size > max {
                max = size;
                count = 1;
            } else if size == max {
                count += 1;
            }
        }

        count
    }
}
